lib/ansible/modules/cloud/vultr/vultr_block_storage.py

```diff
@@ -0,0 +1,241 @@
+'''
+
+EXAMPLES = '''
+- name: Ensure a block storage volume is present
+  local_action:
+    module: vultr_block_storage
+    name: myvolume
+    size: 10
+    region: Amsterdam
+
+- name: Ensure a block storage volume is absent
+  local_action:
+    module: vultr_block_storage
+    name: myvolume
+    state: absent
+'''
+
+RETURN = '''
+---
+vultr_api:
+  description: Response from Vultr API with a few additions/modification
+  returned: success
+  type: complex
+  contains:
+    api_account:
+      description: Account used in the ini file to select the key
+      returned: success
+      type: string
+      sample: default
+    api_timeout:
+      description: Timeout used for the API requests
+      returned: success
+      type: int
+      sample: 60
+    api_retries:
+      description: Amount of max retries for the API requests
+      returned: success
+      type: int
+      sample: 5
+    api_endpoint:
+      description: Endpoint used for the API requests
+      returned: success
+      type: string
+      sample: "https://api.vultr.com"
+vultr_block_storage:
+  description: Response from Vultr API
+  returned: success
+  type: complex
+  contains:
+    attached_to_id:
+      description: The ID of the server the volume is attached to
+      returned: success
+      type: string
+      sample: "10194376"
+    cost_per_month:
+      description: Cost per month for the volume
+      returned: success
+      type: float
+      sample: 1.00
+    date_created:
+      description: Date when the volume was created
+      returned: success
+      type: string
+      sample: "2017-08-26 12:47:48"
+    id:
+      description: ID of the block storage volume
+      returned: success
+      type: string
+      sample: "1234abcd"
+    name:
+      description: Name of the volume
+      returned: success
+      type: string
+      sample: "ansible-test-volume"
+    region:
+      description: Region the volume was deployed into
+      returned: success
+      type: string
+      sample: "New Jersey"
+    size:
+      description: Information about the volume size in GB
+      returned: success
+      type: int
+      sample: 10
+    status:
+      description: Status about the deployment of the volume
+      returned: success
+      type: string
+      sample: "active"
+
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vultr import (
+    Vultr,
+    vultr_argument_spec,
+)
+
+
+class AnsibleVultrBlockStorage(Vultr):
+
+    def __init__(self, module):
+        super(AnsibleVultrBlockStorage, self).__init__(module, "vultr_block_storage")
+
+        self.returns = {
+            'SUBID': dict(key='id'),
+            'label': dict(key='name'),
+            'DCID': dict(key='region', transform=self._get_region_name),
+            'attached_to_SUBID': dict(key='attached_to_id'),
+            'cost_per_month': dict(convert_to='float'),
+            'date_created': dict(),
+            'id': dict(),
```

Comment:
is this by mistake? 2 times key `id`: subid and id?

Reply:
Yep, mistake. Fixed.